packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart

```diff
@@ -0,0 +1,216 @@
+
+        final ProcessResult localNetworkUsage = processManager.runSync(
+          <String>[
+            'plutil',
+            '-extract',
+            'NSLocalNetworkUsageDescription',
+            'xml1',
+            '-o',
+            '-',
+            infoPlistPath,
+          ],
+        );
+        final bool localNetworkUsageFound = localNetworkUsage.exitCode == 0;
+        expect(localNetworkUsageFound, buildMode == BuildMode.debug);
+      });
+
+      testWithoutContext('check symbols', () {
+        final ProcessResult symbols = processManager.runSync(
+          <String>[
+            'nm',
+            '-g',
+            outputAppFrameworkBinary.path,
+            '-arch',
+            'arm64',
+          ],
+        );
+        final bool aotSymbolsFound = (symbols.stdout as String).contains('_kDartVmSnapshot');
+        expect(aotSymbolsFound, buildMode != BuildMode.debug);
+      });
+
+      testWithoutContext('xcode_backend embed_and_thin', () {
+        outputFlutterFramework.deleteSync(recursive: true);
+        outputAppFramework.deleteSync(recursive: true);
+        expect(outputFlutterFrameworkBinary.existsSync(), isFalse);
+        expect(outputAppFrameworkBinary.existsSync(), isFalse);
+
+        final String xcodeBackendPath = fileSystem.path.join(
+          flutterRoot,
+          'packages',
+          'flutter_tools',
+          'bin',
+          'xcode_backend.sh',
+        );
+
+        // Simulate a common Xcode build setting misconfiguration
+        // where FLUTTER_APPLICATION_PATH is missing
+        final ProcessResult xcodeBackendResult = processManager.runSync(
+          <String>[
+            xcodeBackendPath,
+            'embed_and_thin',
+          ],
+          environment: <String, String>{
+            'SOURCE_ROOT': fileSystem.path.join(projectRoot, 'ios'),
+            'BUILT_PRODUCTS_DIR': fileSystem.path.join(
+              projectRoot,
+              'build',
+              'ios',
+              'Release-iphoneos',
+            ),
+            'TARGET_BUILD_DIR': buildPath.path,
+            'FRAMEWORKS_FOLDER_PATH': 'Runner.app/Frameworks',
+            'VERBOSE_SCRIPT_LOGGING': '1',
+            'FLUTTER_BUILD_MODE': 'release',
+            'ACTION': 'install',
+            // Skip bitcode stripping since we just checked that above.
+          },
+        );
+
+        expect(xcodeBackendResult.exitCode, 0);
+        expect(outputFlutterFrameworkBinary.existsSync(), isTrue);
+        expect(outputAppFrameworkBinary.existsSync(), isTrue);
+      }, skip: buildMode != BuildMode.release);
+
+      testWithoutContext('validate obfuscation', () {
+        final ProcessResult grepResult = processManager.runSync(<String>[
+          'grep',
+          '-i',
+          'hello',
+          outputAppFrameworkBinary.path,
+        ]);
+        expect(grepResult.stdout, isNot(contains('matches')));
+      });
+    },
+      skip: !platform.isMacOS,
```

Comment:
Hm, this `group` `skip` didn't seem to work, it ran on Windows and Linux.

Reply:
Oh I know why, it was overridden at the test level https://github.com/dart-lang/test/issues/545